### PR TITLE
Using errexit on build_and_redeploy.sh

### PR DESCRIPTION
**Pull request by andresilva91**
## Motivation

For most cases, if a command fails in a script, the whole script should fail

## Proposal

Using `errexit` (`set -e`) to make that happen. In cases where we want the failure to be allowed, use `|| true`

## Test Plan

Ran the script, works as intended

### CODE REVIEW

**Comment by andresilva91** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1212** <a href="https://app.graphite.dev/github/pr/linera-io/linera-protocol/1212" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/linera-io/linera-protocol/1212?utm_source=stack-comment).

File: kubernetes/linera-validator/build_and_redeploy.sh [hunk 1]
```diff
@@ -38,7 +39,7 @@ handle_options() {
             usage
             exit 0
             ;;
-        --cloud) cloud_mode=1;;
+        --cloud) cloud_mode=1 ;;
```
Comment (andresilva91) [inline kubernetes/linera-validator/build_and_redeploy.sh:42]: @Twey Weirdly, I ran the formatter again, and now it fixed the spacing here 🤣

Comment (Twey) [reply kubernetes/linera-validator/build_and_redeploy.sh:42]: Scary!

File: kubernetes/linera-validator/build_and_redeploy.sh [hunk 2]
```diff
@@ -92,26 +93,26 @@ else
...
 echo "Pods:"
 kubectl get pods
 echo -e "\nServices:"
 kubectl get svc
 
-docker rm linera-test-local
+docker rm linera-test-local || true
 docker run -d --name linera-test-local "$docker_image" &&
     docker cp linera-test-local:wallet.json /tmp/ &&
     docker cp linera-test-local:linera.db /tmp/
```
Comment (Twey) [inline kubernetes/linera-validator/build_and_redeploy.sh:118]:
```suggestion
docker run -d --name linera-test-local "$docker_image"
docker cp linera-test-local:wallet.json /tmp/
docker cp linera-test-local:linera.db /tmp/
```

Comment (Twey) [reply kubernetes/linera-validator/build_and_redeploy.sh:118]: I think this was the statement that originally prompted this PR!  Let's switch this too and then I'm happy :)

**Review by Twey — CHANGES_REQUESTED** [review]

**Review by Twey — APPROVED** [review]

**Pull request merged**